src/main/java/billgates/use_cases/bill_import/BillImportUseCase.java

```diff
@@ -0,0 +1,41 @@
+package billgates.use_cases.bill_import;
+
+import billgates.database.QueryEntryData;
+import billgates.entities.User;
+import billgates.interface_adapters.DatabaseGateway;
+import billgates.use_cases.insert_entry.InsertEntryInputPort;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+/**
+ * Clean Architecture Layer: Application Business Rules
+ * A concrete implementation of the <code>BillImportInputPort</code>.
+ *
+ * @author Eva
+ * @see BillImportInputPort
+ */
+
+public class BillImportUseCase implements BillImportInputPort {
+    private final DatabaseGateway gateway;
+
+    public BillImportUseCase(DatabaseGateway gateway) {
+        this.gateway = gateway;
+    }
+
+    @Override
+    public void importBill(BillImportRequestModel model) {
+        List<QueryEntryData> result = this.gateway.getBillData(User.getInstance().getCurrentBillID()).getEntries();
```

Comment:
What is the purpose of this line, why do we need to get the bill data when we are inserting entries?

Alos, if you decide to include this line, maybe we can remove duplicated entries and not insert them again? Maybe you can check if there are duplicated `Value` and `Date` to check if the imported entry is duplicated or not.